src/core/components/config.js

```diff
@@ -42,13 +25,7 @@ module.exports = function config (self) {
         return callback(new Error('Invalid value type'))
       }
```

Comment:
It should. https://github.com/ipfs/js-ipfs-repo/pull/140 supports this now (via https://github.com/ipfs/js-ipfs-repo/commit/8e8055ddae3f4f6b44ebb4f8932393dc95327039).
Fixed.